### ISSUE TITLE
nextjs 13.4.13 is not transpiling code to browser old versions even with browserslist option in package.json

### ISSUE THREAD

**Issue by yanv1991** [vercel/next.js#53821]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 20.6.0: Thu Mar  9 20:39:26 PST 2023; root:xnu-7195.141.49.700.6~1/RELEASE_X86_64
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.14-canary.1
      eslint-config-next: 13.4.13
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/yanv1991/test-2017

### To Reproduce

run `yarn build` and then `yarn start`

`note: webpack custom config was added to disable minimizer`

look for the `.next` folder inside the `main-xxxxx` file you will see non es5 syntax like: 

![image (1)](https://github.com/vercel/next.js/assets/3800032/87972aea-f486-4ed1-8402-7084fe67519f)

### Describe the Bug

when adding `browserslist` option in package.json with a value like this: `>0.3%, defaults, supports es5` the next bundle is still generating modern syntax.

### Expected Behavior

if browserslist with es5 target is configured the bundle should not generate modern syntax like `let, const`....

### Which browser are you using? (if relevant)

Chrome 48.0.2527.0

### How are you deploying your application? (if relevant)

local

**Comment by EloB** [comment]
I've tried older versions and seems like it doesn't work there either.

**Comment by EloB** [comment]
My problem was `transpilePackages`. I had some packages in `node_modules/` that wasn't transpiled. So maybe not related.

**Comment by yanv1991** [comment]
@EloB  yes the repo that I posted here is a app generated using the next-create-app CLI, without any dependency and the error looks like is inside nextjs core

**Comment by EloB** [comment]
@yanv1991 Have you tried babel? It might be a SWC issue.

**Comment by yanv1991** [comment]
yes actually the repo uses babel.config

**Comment by EloB** [comment]
@yanv1991 Does `next` contain new syntax? :)

**Comment by EloB** [comment]
Have you tried if this works with SWC? I really don't want to use babel.

**Comment by yanv1991** [comment]
> @yanv1991 Does `next` contain new syntax? :)

yes, take a look to the error, this is running in chromium 48:
![](https://files.slack.com/files-pri/T01U2MF6PDZ-F05M8LWK8TG/image.png)

**Comment by yanv1991** [comment]
> Have you tried if this works with SWC? I really don't want to use babel.

it worked even with swc, as long I keep `next` in the transpilePackage

**Comment by woshiqiang1** [comment]
This bug sames occur after v13.2.0,  use v13.2.0 the main-**.js is ok, but when you import third part package in the project the file in dist chunks folder still have  es6 syntax even you config browserslist.

**Comment by yanv1991** [comment]
@woshiqiang1  for 3rd parties you need transpilePackages option

**Comment by EloB** [comment]
@woshiqiang1 The easiest way to find with package it was is:
**next.config.js**
```js
module.exports = {
  webpack: (config) => {
    config.optimization.minimize = false;
    return config;
  },
}
```

Sometimes it might be a sub dependency. When you do like that you still have the comments left and discovered that sub packages contained ES6 code. Than I just added them too.

**Comment by woshiqiang1** [comment]
@EloB Thanks I try the way you said above and it works !! but it's really not a very elegant and convenient way hope Next.js optimize this DX someday.

**Comment by longzheng** [comment]
I also ran into issues after upgrading to Next 13.5.2, the build would generate non-ES5 syntax (that I check with the [es-check package](https://github.com/yowainwright/es-check) `es-check es5 '.next/static/chunks/main-*.js'`)

I also found that the output chunk contained some Next.js router code that had `let`.

I also fixed it by adding `next` to my `transpilePackages` config.

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ['next'],
}
 
module.exports = nextConfig
```

**Comment by egemon** [comment]
13.5.6 same issue.  Adding next breaks the build

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ['next'],
}
 
module.exports = nextConfig
```

Error tail
```
> next build

error - No Sentry auth token configured. Source maps will not be uploaded.
You can find information on how to generate a Sentry auth token here: https://docs.sentry.io/api/auth/
After generating a Sentry auth token, set it via the SENTRY_AUTH_TOKEN environment variable during the build.

Failed to compile.

./node_modules/swr/_internal/dist/index.mjs
Attempted import error: 'useEffect' is not exported from 'react' (imported as 'useEffect').

Import trace for requested module:
./node_modules/swr/_internal/dist/index.mjs
./node_modules/swr/core/dist/index.mjs

```

**Comment by tua-Mascot** [comment]
I conducted an experiment on a pure next.js project with these basic settings.
Windows 10
Node.js v20.9.0
![Code_bDSpNZEJ3T](https://github.com/vercel/next.js/assets/28605148/4aef4c25-d85f-4219-8d38-63337eb0730d)

It seems that the settings for browserslist inside package.json have no effect on the size of the bundle and have a very strange effect on the size of the .next folder.

clean project (npx create-next-app@latest)
```
Route (app)                              Size     First Load JS
┌ ○ /                                    5.12 kB        89.3 kB
└ ○ /_not-found                          885 B            85 kB
+ First Load JS shared by all            84.2 kB
  ├ chunks/69-1b6d135f94ac0e36.js        28.9 kB
  ├ chunks/fd9d1056-cc48c28d170fddc2.js  53.4 kB
  └ other shared chunks (total)          1.86 kB

.next folder size - 32.4 MB (33,984,512 bytes)
```

"browserslist": ["last 2 Chrome versions"] (package.json)
```
Route (app)                              Size     First Load JS
┌ ○ /                                    5.12 kB        89.3 kB
└ ○ /_not-found                          885 B            85 kB
+ First Load JS shared by all            84.2 kB
  ├ chunks/69-1b6d135f94ac0e36.js        28.9 kB
  ├ chunks/fd9d1056-cc48c28d170fddc2.js  53.4 kB
  └ other shared chunks (total)          1.86 kB

.next folder size - 31.4 MB (32,952,320 bytes)
```

"browserslist": ["last 2 Chrome versions"] (package.json) + transpilePackages: ['react', 'react-dom', 'next'] (next.config.mjs)
```
Route (app)                              Size     First Load JS
┌ ○ /                                    5.12 kB        89.3 kB
└ ○ /_not-found                          885 B            85 kB
+ First Load JS shared by all            84.2 kB
  ├ chunks/69-1b6d135f94ac0e36.js        28.9 kB
  ├ chunks/fd9d1056-cc48c28d170fddc2.js  53.4 kB
  └ other shared chunks (total)          1.86 kB

.next folder size - 33.0 MB (34,631,680 bytes)
```

Can one of the developers look at this problem (possibly - @timneutkens @Timer @leerob)?
Looks like a serious bug (but maybe I just don’t fully understand how it should work 😅).

**Comment by tua-Mascot** [comment]
I also noticed an extremely strange effect. 
If I remove browserslist from package.json (I think that the default settings specified in the documentation should be used), then in the chunk file ".next\static\chunks\app\_not-found-[hash].js" in default next.js 14 project I see let variables. If I add browserslist with “last 2 Chrome versions”, then in the same chunk I see var variables 👀.
In general, there is a feeling that no matter what target you set in browserslist and no matter what experimental settings you use (for example, from this issue - https://github.com/vercel/next.js/issues/12557), the size of the client bundle does not change in any way (no difference is observed in either the CLI report after the 'build' command or in the 'next/bundle-analyzer' report. sometimes by 1-2 kb, but even then rarely).
It's quite a frustrating experience. I don't want to sound rude (and I apologize if it sounds like that), but I see only 2 possible options here:
1) I don’t understand how this should work in complex systems like Next.js
2) We really have a problem with the target for the final bundle and tree-shaking, and judging by the number of issues and comments inside - this is a long-standing problem that for some reason is not being solved or does not receive enough attention from Next.js team.

**Comment by jim-king-2000** [comment]
I tried adding next to my transpilePackages config.
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ['next'],
}
 
module.exports = nextConfig
```
But there are still private properties in my js. Is it possible to transpile private properties?

**Comment by marcoshernanz** [comment]
Thanks for the report. I confirmed this is the same browserslist transpilation gap tracked in #45473. Next.js leaves syntax newer than the configured browser targets in the client bundle. Closing as a duplicate so the compiler compatibility investigation stays consolidated.
